todo-list-backend/src/index.ts

```diff
@@ -0,0 +1,124 @@
+import {decode, sign} from "jsonwebtoken";
+import {applyMiddleware} from "graphql-middleware";
+import {ApolloServer, gql} from "apollo-server";
+import {makeExecutableSchema} from "graphql-tools";
+
+import {User} from "./types";
+import {InMemoryStorage, IStorage} from "./data"
+import {buildPermissions} from "./permissions";
+
+const typeDefs = gql`
+    type User {
+        name: String
+    }
+
+    type ToDo {
+        id: Int
+        name: String
+        done: Boolean
+        assignee: User
+    }
+
+    type Query {
+        readTodos: [ToDo]!
+    }
+
+    type Mutation {
+        login(name: String!, password: String!): String
+        createTodo(name: String!): ToDo
+        updateTodoName(id: Int!, name: String!): ToDo
+        updateTodoDone(id: Int!, done: Boolean!): ToDo
+        deleteTodo(id: Int!): ToDo
+        createUser(name: String! password: String!): User
+    }
+`;
+
+function buildResolvers(storage: IStorage, authSecret: string): any {
+    return {
+        Query: {
+            readTodos: (parent: any, args: any, context: any, info: any) => {
+                let user: User = context.user;
```

Comment:
If you want, you can use object destructuring here:
```suggestion
                let { user } = context
```
Saves you one `.user`.

Reply:
Sadly this would remove the type annotation :disappointed:. According to a stackoverflow post  `let { user }: {user: User} = context` is possible but I'am not sure if this is much better than the plain old `context.user`.